src/general/horizontalpodautoscaler/src.rego

```diff
@@ -0,0 +1,43 @@
+package k8shorizontalpodautoscaler
+
+violation[{"msg": msg}] {
+  input.review.kind.kind == "HorizontalPodAutoscaler"
+  hpa := input.review.object
+
+  not input_replica_limit(hpa)
+  msg := sprintf("The %v <%v> minReplicas %v or maxReplicas %v is not allowed: %v. Allowed ranges: %v", [hpa.kind, hpa.metadata.name, hpa.spec.minReplicas, hpa.spec.maxReplicas, input.parameters.ranges])
+}
+
+violation[{"msg": msg}] {
+  input.review.kind.kind == "HorizontalPodAutoscaler"
+  hpa := input.review.object
+
+  not input_replica_spread(hpa)
+  
+  msg := sprintf("The %v <%v> is configured with minReplicas %v and maxReplicas %v which is a spread of %v replica(s). The spread must be at least %v replica(s)", [hpa.kind, hpa.metadata.name, hpa.spec.minReplicas, hpa.spec.maxReplicas, hpa.spec.maxReplicas - hpa.spec.minReplicas, input.parameters.minimumReplicaSpread])
+}
+
+violation[{"msg": msg}] {
+  input.review.kind.kind == "HorizontalPodAutoscaler"
+  hpa := input.review.object
+  input.parameters.enforceScaleTargetRef
+  
+  not data.inventory.namespace[hpa.metadata.namespace][hpa.spec.scaleTargetRef.apiVersion][hpa.spec.scaleTargetRef.kind][hpa.spec.scaleTargetRef.name]
```

Comment:
If the scaleTargetRef kind has different casing than what is considered to be standard, the policy will consider it a violation due to case sensitivity of the keys in OPA. For example, an HPA with scaleTargetRef.kind=statefulset when it exists in GK cache under the key **S**tateful**S**et. They are both valid from the perspective of the HPA, I don't know whether there is any solution to this.

Reply:
probably you'd have to iterate over all GVKs in `data.inventory`, setting the Kind, etc. to variables, then doing a case-insensitive comparison via `strings.lower()`.

The downside of this is that it may be less performant, depending on the contents of data.inventory.